python/ray/util/debugpy.py

```diff
@@ -59,9 +60,34 @@ def _ensure_debugger_port_open_thread_safe():
 
         debugger_port = ray._private.worker.global_worker.debugger_port
         if not debugger_port:
-            (host, port) = debugpy.listen(
-                (ray._private.worker.global_worker.node_ip_address, 0)
+            # Define the port range to use
+            port_range_start = int(
+                os.environ.get("RAY_DEBUGGER_PORT_RANGE_START", "52000")
             )
+            port_range_end = int(os.environ.get("RAY_DEBUGGER_PORT_RANGE_END", "52100"))
+
+            # Create a list of ports in the range and shuffle it
+            available_ports = list(range(port_range_start, port_range_end + 1))
+            random.shuffle(available_ports)
+
+            # Try ports in random order from the range
+            node_ip = ray._private.worker.global_worker.node_ip_address
```

Comment:
Introducing this variable doesn't add value, it is probably better to remove it :)